File: Gregs2/Models/Car.cs
using Gregs2.interfaces;
namespace Gregs2.Models;

public class Car : IHasCreator
{
  public string Make { get; set; }
  public string Model { get; set; }
  public int Year { get; set; }
  public decimal Price { get; set; }
  public string Description { get; set; }
  public string ImgUrl { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public bool IsSold { get; set; }
}
public class FavoritedCar : Car
{
  public int FavoriteId { get; set; }
}

File: Gregs2.client/src/models/Car.js
export class Car {
  constructor(data) {
    this.id = data.id
    this.make = data.make
    this.model = data.model
    this.year = data.year
    this.price = data.price
    this.description = data.description
    this.imgUrl = data.imgUrl
    this.creatorId = data.creatorId
    this.creator = data.creator
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.isSold = data.isSold
  }
}